test/test_hyperparameters.py

```diff
@@ -328,6 +326,176 @@ def test_normalfloat_to_integer(self):
         f2_actual = f1.to_integer()
         self.assertEqual(f2_expected, f2_actual)
 
+    def test_betafloat(self):
+        # TODO test non-equality
+        f1 = BetaFloatHyperparameter("param", lower=-2.0, upper=2.0, alpha=3.0, beta=1.0)
+        f1_ = BetaFloatHyperparameter("param", lower=-2, upper=2, alpha=3, beta=1)
+        self.assertEqual(f1, f1_)
+        self.assertEqual(
+            "param, Type: BetaFloat, Alpha: 3.0 Beta: 1.0, Range: [-2.0, 2.0], Default: 2.0",
+            str(f1_))
+
+        # test parameters that do not create a legit beta distribution
+        with self.assertRaises(ValueError):
+            BetaFloatHyperparameter("param", lower=-2, upper=2, alpha=-11, beta=5)
+            BetaFloatHyperparameter("param", lower=-2, upper=2, alpha=5, beta=-11)
+
+        # test parameters that do not yield a finite co-domain
+        with self.assertRaises(ValueError):
+            BetaFloatHyperparameter("param", lower=-2, upper=2, alpha=0.5, beta=11)
+            BetaFloatHyperparameter("param", lower=-2, upper=2, alpha=11, beta=0.5)
+
+        u1 = UniformFloatHyperparameter("param", lower=0.0, upper=1.0)
+        b1 = BetaFloatHyperparameter("param", lower=0.0, upper=1.0, alpha=3.0, beta=1.0)
+
+        # with identical domains, beta and uniform should sample the same points
```

Comment:
@mfeurer this comment is a better description about my comment above for `get_neighbours` and it's behaviour.